app/helpers/application_helper/button/ems_infra_recheck_auth_status.rb

```diff
@@ -0,0 +1,5 @@
+class ApplicationHelper::Button::EmsInfraRecheckAuthStatus < ApplicationHelper::Button::Basic
+  def skip?
+    !@record.is_available?(:authentication_status)
+  end
+end
```

Comment:
You don't need to create several classes with the same code. 

If the behavior you need here just checks for auth status on the record, you can just create 1 class for that and share the class for all the buttons